### PR TITLE
Handle 429 upon upload to SCC

### PR DESCRIPTION
**Pull request by mbelur**
When the put request to SCC fails with 429,
by default, we exit and do not retry.
However if the option --retry-on-rate-limit
is set, then the program waits for
the period specified in the response
header Retry-After from the SCC API and retry again.

### CODE REVIEW

File: src/scc_hypervisor_collector/api/uploader.py [hunk 1]
```diff
@@ -81,3 +88,31 @@ def check_creds(self,
                                 headers=self.headers,
                                 allow_redirects=False)
         return response.status_code == 200
+
+    def retry_upload(self, delay: int, path: str, headers: dict,
```
Comment (rtamalin) [inline src/scc_hypervisor_collector/api/uploader.py:92]: So you are duplicating the request call here which works, but may not be the best code pattern; if we need to tweak anything to do with the call we have to do it in both places...

Potentially you could have this routine handle the try & retry of the request call itself, e.g. default the delay to 0, and make it the last argument, and only if delay is non-zero do you log the info message and call time.sleep()

File: tests/unit/test_uploader.py [hunk 2]
```diff
@@ -128,13 +128,13 @@ def test_upload_hypervisor_details_to_scc_retry(self, collected_results):
                                         url=scc_url))
 
         uploader = SCCUploader(scc_creds)
-
-        with mock.patch('requests.put', return_value=FakeResponse(429)
+        with mock.patch('requests.put', return_value=FakeResponse(429, {'Retry-After': 5})
```
Comment (rtamalin) [inline tests/unit/test_uploader.py:131]: Probably could reduce this to 1 second delay, so as to not increase unit test run time unnecessarily...

Note also that time.sleep can take a floating point seconds value to sleep for, so you could actually switch to the retry_upload() taking a float or int as possible values, and use 0.001 here, and get the same effect in terms of log message getting generated, without having to wait for any real time having passed ;-)

File: tests/unit/test_uploader.py [hunk 3]
```diff
@@ -8,7 +8,7 @@
     SCCUploader
 )
 
-FakeResponse = namedtuple('FakeResponse', 'status_code')
+FakeResponse = namedtuple('FakeResponse', 'status_code, headers')
```
Comment (rtamalin) [inline tests/unit/test_uploader.py:11]: Note that we could change this to 
```suggestion
class FakeResponse:
    def __init__(self, status_code, headers=None):
        if headers is None:
            headers = {}
        self.status_code = status_code
        self.headers = headers
```
And then you wouldn't need to change any of the FakeResponse calls in the rest of the code unless you wanted to include header fields.

File: src/scc_hypervisor_collector/api/uploader.py (hunk 1 above)
Comment (mbelur) [reply src/scc_hypervisor_collector/api/uploader.py]: Removed duplication of the request call in the latest changes.

File: tests/unit/test_uploader.py (hunk 2 above)
Comment (mbelur) [reply tests/unit/test_uploader.py]: Set the delay to 1 sec

File: tests/unit/test_uploader.py (hunk 3 above)
Comment (mbelur) [reply tests/unit/test_uploader.py]: updated with the suggested changes

**Pull request merged**